### PR TITLE
Fix vim

### LINKED ISSUE #86: [Bug report] Plugin and color schemes broken in Vim after "Rework"
## Describe the bug

After updating the plugin to the newest state after the rework was merged leads to errors at startup of Vim:

```
Error detected while processing [..]/vim/pack/ptm/start/base16-vim/plugin/tinted-vim.vim:
line    8:
E319: Sorry, the command is not available in this version: lua require('tinted-vim')
```

Furthermore, removing the `tinted-vim` plugin (that seems to require Neovim anyway), leads to the following errors with some of the updated colour schemes (`base16-material` in this case):

```
Error detected while processing [..]/.vimrc[233]..[..]/vim/pack/ptm/start/base16-vim/colors/base16-material.vim[277]..function <SNR>3_hi[1]..Tinted_Hi:
line   34:
E254: Cannot allocate color #none
E254: Cannot allocate color #none
E254: Cannot allocate color #underline
Error detected while processing /Users/patthemav/.vimrc[253]../opt/homebrew/Cellar/vim-custom/9.1.0850/share/vim/vim91/syntax/syntax.vim[20]../opt/homebrew/Cellar/vim-custom/9.1.0850/share/vim/vim91/syntax/synload.vim:
line   20:
E185: Cannot find color scheme 'base24-material'
```

By the looks of it, this updated theme uses `none` and `underline` as the `guisp` argument to `Tinted_Hi`, which expects a supported colour value instead.

The updated colour scheme also contains the following comment:

```
" `base16_colorspace` and `base16colorspace` are legacy properties and
" exist to keep existing setups from breaking
```

But the actual code only contains a fallback conditional for `base16_colorspace` and _not_ `base16colorspace`, so any users running with the second value will actually experience breakage.

And last, but not least, it is an error that the scheme bears the colour name `base24-material`, when it is indeed `base16-material`.

I spot-checked some other base16-based color schemes and they all exhibited the same issues.

## Expected behavior

* The plugin should not load code that requires Neovim by default and continue to work in Vim if it is advertised as a Vim plugin.
* The colour schemes should work out of the box and not contain syntax errors
* The colour schemes' colour names should match their file names
* The colour schemes thus should load without errors

## **System**

**Vim or Neovim:** Vim

**Vim or Neovim version:** Vim 9.1.0850

## Minimal configuration file

This assumes using Vim's built-in plugin support by placing `tinted-vim` in `~/pack/plugins/start/tinted-vim`:

```
set nocompatible

syntax enable

let tinted_colorspace=256
colorscheme base16-material
```

### PR DESCRIPTION
**Pull request by GordianDziwis**
## Description

Fixes #86.

What is still an issue, is that Vim defines other Default highlights than Nvim. Check `ErrMsg` highlight as an example.

Should we clear the used highlights by default, so that we start from a clean state? @JamyGolden?

## Checklist

- [ ] I have included the built files `./colors/*.vim` in a separate commit and followed [the build instructions](https://github.com/tinted-theming/tinted-vim/blob/main/CONTRIBUTING.md)
- [ ] I have confirmed that my changes produce no regressions after building

### CODE REVIEW

File: colors/base16-apathy.vim [hunk 1]
```diff
@@ -131,6 +131,22 @@ else
...
+  let s:cterm13        = s:cterm0A
+  let g:tinted_cterm13 = s:cterm0A
+  let s:cterm14        = s:cterm0B
+  let g:tinted_cterm14 = s:cterm0B
+  let s:cterm15        = s:cterm0C
+  let g:tinted_cterm15 = s:cterm0C
+  let s:cterm16        = s:cterm0D
+  let g:tinted_cterm16 = s:cterm0D
+  let s:cterm17        = s:cterm0E
+  let g:tinted_cterm17 = s:cterm0E
```
Comment (GordianDziwis) [inline colors/base16-apathy.vim:149]: @JamyGolden Are those correct?

**Comment by GordianDziwis** [comment]
@JamyGolden 

- Are the cterm fallbacks for base24 correctly defined?
- Vim does define other default highlight than Nvim, see `ErrMsg` highlight. Should we just clear the highlights by default?

**Comment by JamyGolden** [comment]
I don't see any `ErrMsg`, you might mean `ErrorMsg` but I don't see anything noticable about that line https://github.com/tinted-theming/tinted-vim/pull/89/files#diff-8e4da39bbefb342dbac24c7b8fe153c9e2b12798d48a2ea5a75721d462ed88faR304

Clearing highlights would make our job easier, but I'm worried it might break things unexpectedly. For example changing a theme with `:colorscheme base16-oceanicnext` would remove highlights added by plugins for example, unless I'm mistaken.

**Comment by GordianDziwis** [comment]
> I don't see any `ErrMsg`, you might mean `ErrorMsg` but I don't see anything noticable about that line https://github.com/tinted-theming/tinted-vim/pull/89/files#diff-8e4da39bbefb342dbac24c7b8fe153c9e2b12798d48a2ea5a75721d462ed88faR304

Tinted-vim sets the foreground of `ErrorMsg` red, and vim sets the background red, which makes `ErrorMsg` unreadable.

> Clearing highlights would make our job easier, but I'm worried it might break things unexpectedly. For example changing a theme with `:colorscheme base16-oceanicnext` would remove highlights added by plugins for example, unless I'm mistaken.

We would only clear the highlights set by tinted-vim, so highlights set by plugins should be untouched (unless we set them explicitly).

**Comment by JamyGolden** [comment]
Ah ok, yeah clearing the highlights we work with makes sense.

**Comment by PatTheMav** [comment]
Just checked the PR:

* The themes still don't support the old `base16colorspace` variable, even though the comment block above the fallback code makes explicit mention of this legacy variable (which is the original variable used to be enable this functionality)
* If I'm not mistaken, this following block in the plugin code is a guard against multiple initialisation:

```vim
if exists('g:tinted-vim')
    finish
endif
let g:loaded_yourplugin = 1
```

The `tinted-vim` global is never set in my tests (is it set by the Lua code). Also `loaded_yourplugin` sounds like the default variable used in some vim plugin template and should be changed to reflect the name of the plugin.
* The internal name of `base16-material` is `base24-material`. Is this intended? Because it still needs to be set as `base16-material` to take effect.

As I'm using colours set in the shell (kinda like `base16-shell` the different themes all look the same visually in my tests, but I did not encounter any vim errors while switching between them.

**Comment by GordianDziwis** [comment]
Thanks a lot for the review!

>     * The themes still don't support the old `base16colorspace` variable, even though the comment block above the fallback code makes explicit mention of this legacy variable (which is the original variable used to be enable this functionality)

Fixed, was a typo.

>     * If I'm not mistaken, this following block in the plugin code is a guard against multiple initialisation:
> 
> 
> ```viml
> if exists('g:tinted-vim')
>     finish
> endif
> let g:loaded_yourplugin = 1
> ```
> 
> The `tinted-vim` global is never set in my tests (is it set by the Lua code). Also `loaded_yourplugin` sounds like the default variable used in some vim plugin template and should be changed to reflect the name of the plugin.

You are right was an artifact from some template, I removed it.

>     * The internal name of `base16-material` is `base24-material`. Is this intended? Because it still needs to be set as `base16-material` to take effect.

Good catch!

> As I'm using colours set in the shell (kinda like `base16-shell` the different themes all look the same visually in my tests, but I did not encounter any vim errors while switching between them.

Without `set termguicolors` the colors are defined by the terminal, so they should not change. I updated the README.md.

**Pull request merged**